src/Serializer/DataArraySerializer.php

```diff
@@ -46,6 +46,6 @@ public function item($resourceKey, array $data)
      */
     public function null()
     {
-        return ['data' => []];
+        return [$resourceKey ?: 'data' => []];
```

Comment:
`$resourceKey` is used in other `Serializer`, e.g. `ArraySerializer` or `JsonSerializer`. `DataSerializer` uses the hardcoded `data`. That's why it was called `DataSerializer`.
Use another Serializer or write your own if you want to change this behavior.